torch/_prims/nvfuser_executor.py

```diff
@@ -268,11 +268,42 @@ def __call__(self, *args):
         )
 
 
+def _remove_empty_like_fill(gm: GraphModule):
+    # Remove empty_like + fill nodes that prevent lowering to nvprims
+    # This is a workaround for nonoptimal traces of C++ code `(1 - tensor)`
+    # https://github.com/pytorch/pytorch/issues/86612
+
+    # Here when we see a `sub` node, we check if the first input is a result of
+    # filling a tensor with a scalar
+    # If so, we replace the first argument of the `sub` node with a scalar
+    for node in gm.graph.nodes:
+        if node.op == "call_function":
+            if (
+                node.target == torch.ops.aten.sub.Tensor
```

Comment:
for `torch.ops.aten.sub` replacing a tensor arg with scalar arg changes type promotion semantics, for `nvprims` it's ok since nvprims ops are not type promoting.

Reply:
Thanks for pointing this out! I think I need this change only for nvprims.sub.default. I will remove modifying `aten.sub` and modify the test.